### ISSUE TITLE
[UX] Improve the "URL language detection configuration" form (and fix duplicate help text)

### ISSUE THREAD

**Issue by klonos** [backdrop/backdrop-issues#3449]
**Describe the bug**
The form in `/admin/config/regional/language/detection/url` has a duplicate portion of text when selecting the "Domain" option...

**Steps To Reproduce**
To reproduce the behavior:
1. Enable the Locale module.
2. Navigate to `/admin/config/regional/language/detection/url`
3. Select the "Domain" option from the "Part of the URL that determines language" radios...

**Actual behavior**

<img width="972" alt="screen shot 2018-12-30 at 7 01 16 am" src="https://user-images.githubusercontent.com/2423362/50544632-72afbb80-0c05-11e9-9ae8-6699ffd325ac.png">

**Expected behavior**

- The help text should not be duplicated.
- The bolded text should be a proper inline warning instead.
- Each of the radio options should have short individual help text with examples (one currently needs to toggle between the two options and read the extended help text to understand what each option does).

---
PR by @klonos: ...

**Comment by klonos** [comment]
PR up for review:

<img width="951" alt="screen shot 2018-12-30 at 7 31 52 am" src="https://user-images.githubusercontent.com/2423362/50544678-3f6e2c00-0c07-11e9-83b6-61385c110394.png">

**Comment by olafgrabienski** [comment]
@klonos I've had a look at the sandbox site, which looks good. Two suggestions:

- We don't need bold text in the warning message, as it's already styled as a warning.
- In the *Path prefix* configuration help text example, change "deutsch" to "de". I guess, "de" is more common, and apart of that, "de" is also used in the help text at the *Path prefix* radio button.

**Comment by BWPanda** [comment]
Made a PR (based on (https://github.com/backdrop/backdrop/pull/2437)) which addresses @olafgrabienski's second point (@klonos had already addressed the first in his PR): https://github.com/backdrop/backdrop/pull/2756

**Comment by olafgrabienski** [comment]
@BWPanda Thanks for addressing my 2nd point (change "deutsch" to "de"), this looks good now in the sandbox site!

---

While I was on it, I've had a look on both options (Path prefix vs. Domain) and saw that there are a few textual inconsistencies (see the bold parts):

**Path prefix:**
> Language codes or other custom text to use as a path prefix **for URL language detection**.
> For the default language, this value **may be left blank**.
> Example: Specifying de as the path prefix code for German **results in URLs** like https://example.com/de/contact.

**Domain:**
> The domain names to use **for these languages**.
> **Leave blank** for the default language.
> Example: Specifying de.example.com as language domain for German **will result in an URL** like https://de.example.com/contact.

Shall we address them right away here, or better open a separate issue?

**Comment by BWPanda** [comment]
Easy enough to do here I think.

**Comment by klonos** [comment]
Sorry for failing to chime in earlier, but I don't thing that the "deutsch" to "de" change is required. Please note the bolded bit on the help text:

> Language codes or **other custom text** to use as a path prefix...

This is not limited to language codes only. Using "deutsch" served as an example to demonstrate that custom text can be used too. For all I know, people can configure their site to be: https://example.com/klingon/contact

**Comment by klonos** [comment]
...perhaps actually using "klingon" in the example URL would help make the point clearer 😆

**Comment by BWPanda** [comment]
Hmm, good point @klonos. Re-reading both descriptions texts, I don't see why either can't just be any string. E.g. Domain: `myAwesomeLanguage.example.com` and Prefix: `example.com/fakeLanguage`

And yes, feel free to use your original PR as I named mine wrongly. Also please address @olafgrabienski's [comments](https://github.com/backdrop/backdrop-issues/issues/3449#issuecomment-506746953) re. the description differences, then we should be good to go.

**Comment by olafgrabienski** [comment]
> I don't thing that the "deutsch" to "de" change is required. (...) This is not limited to language codes only, and using "deutsch" served as an example to demonstrate that custom text can be used too.

Yes, other strings than language codes are possible but why emphasize this use case? I'd prefer to show language codes in the example as they are more common.

**Comment by klonos** [comment]
> ...but why emphasize this use case? I'd prefer to show language codes in the example as they are more common.

Yes, the way I see this form work is that we state the obvious, most common use case in the help text in the radio buttons; then further down, we communicate that the prefix does not necessarily need to be a language code - any arbitrary text is supported:

![Screen Shot 2019-06-29 at 9 33 37 am](https://user-images.githubusercontent.com/2423362/60376391-0ac21c80-9a52-11e9-83a1-e4e36c46f50a.png)

**Comment by klonos** [comment]
...writing the above gave me ideas on how to tweak the description for clarity:

![Screen Shot 2019-06-29 at 10 33 15 am](https://user-images.githubusercontent.com/2423362/60377349-53310880-9a59-11e9-9849-0cec2b72c223.png)

I have also moved things around, to have less repetition of translatable strings in the code (less work for our translators).

**Comment by klonos** [comment]
...the second bullet point sounds like a "did you know?" thing to me, which is a perfect match for the reason #3155 was implemented for. So things could look like this:

![Screen Shot 2019-06-29 at 10 43 44 am](https://user-images.githubusercontent.com/2423362/60377484-dacb4700-9a5a-11e9-8ef9-2227f978ce73.png)

I will wait for feedback before I update the PR to do that though.

**Comment by BWPanda** [comment]
I like it

**Comment by matthui** [comment]
Looks good, but for the path prefix configuration shouldn't the example be deutsch.example.com/contact?

**Comment by matthui** [comment]
Also, although I like the popups in the added image, I feel like its a little distracting to have more attention-grabbing popups (both the warning and the info) than text, though both are equally important.  The current one with bullet points works and seems simple enough.

**Comment by klonos** [comment]
> ...but for the path prefix configuration shouldn't the example be deutsch.example.com/contact?

Thanks for the review @matthui 👍 ...and good catch re the domain prefix. Looking for a smart way to fix this without increasing the burden for translators.

**Comment by klonos** [comment]
...the label "pr - needs code review" should be retained @matthui; unless you did code-review it, in which case, you should also add the "pr - ready to be committed" label 😉

**Comment by matthui** [comment]
@klonos I did code review it, but it seemed like you were planning to add more changes to it so I didn't add the ready to be committed.

**Comment by olafgrabienski** [comment]
@klonos Thanks for the clarification in https://github.com/backdrop/backdrop-issues/issues/3449#issuecomment-506906307, agreed! From the following suggestions I like https://github.com/backdrop/backdrop-issues/issues/3449#issuecomment-506911932 (with bullet points / without info message) the most. IIRC, this is also the state of the current PR in the sandbox.

Speaking of the sandbox, I noticed an inconsistency between the *Path prefix* and the *Domain* configuration text. Not sure if you left it by accident:

> Text to use as a path prefix **for URL language detection**.
vs.
> The domain names to use **for these languages**.

Furthermore, in the *Domain configuration* text, the text of the 2nd bullet point refers to path configuration:

> For example, specifying the full name of the language (deutsch) as the **path prefix** code for German, results in URLs like https://**example.com/deutsch/contact**.

**Comment by klonos** [comment]
@olafgrabienski thanks for the second pair of 👀 👍 ...see my responce to @matthui above ☝️:

> ..and good catch re the domain prefix. Looking for a smart way to fix this without increasing the burden for translators.

It seems that I may need to revert to separate texts, but that would also revert the benefit of making things easier for translators. Still thinking about it 🤔

**Comment by docwilmot** [comment]
I think the existing PR covers the original issue. Is there any plan to update it any further? It looks good to me otherwise.

**Comment by docwilmot** [comment]
@klonos update please?

**Comment by klonos** [comment]
Sorry for not providing an update for this long. PR rebased and updated, UI simplified:

![image](https://user-images.githubusercontent.com/2423362/149622876-0e2c5948-43c1-4c38-bb87-09f0922eac3f.png)

- Basic examples added below each of the options
- Note about the value of the default language allowed to be blank moved from the fieldset description as description in the field for the default language instead.
- All fields besides the one for the default language made required (this will be further improved once we have #5348 merged)

**Comment by klonos** [comment]
...actually need to fix some failing tests first.
